### PR TITLE
Standardization for carenet_health ads script

### PR DESCRIPTION
**Pull request by raaayden**
Update :
1. Remove subqueries and nested CTE
2. Update format
3. Changes the flow for more optimize and remove redundancy

Left is new script Right is old script to compare no of rows:

1. x-marketing.carenet_health.linkedin_ads_performance
image
2. x-marketing.carenet_health.all_ads_performance
image

### CODE REVIEW

File: Clients/Carenet/Normalization Script/Ads/carenet_health_ads_performance.sql
Comment (izztsfn) [inline Clients/Carenet/Normalization Script/Ads/carenet_health_ads_performance.sql:1]: Formatting wise, all good for this one

File: Clients/Carenet/Normalization Script/Ads/carenet_health_all_ads_perf.sql [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+    activity.conversions AS _conversions, 
+    NULL AS _leads,
+    NULL AS _landingpageclick
+  FROM `x-marketing.carenet_health.google_search_campaign_performance` activity
+  JOIN `x-marketing.carenet_google_ads.campaigns` campaign 
+    ON campaign.id = activity.campaign_id
+),
+_google_sem AS (
+  SELECT DISTINCT *
+  FROM (
```
Comment (izztsfn) [inline Clients/Carenet/Normalization Script/Ads/carenet_health_all_ads_perf.sql:73]: Create a new CTE for this to remove subquery

File: Clients/Carenet/Normalization Script/Ads/carenet_health_all_ads_perf.sql [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+  _all.*, 
+  CASE 
+    WHEN _end_date IS NOT NULL 
+    THEN CONCAT(CAST(_start_date AS DATE),' ','-',' ',CAST(_end_date AS DATE)) 
+    WHEN _start_date IS NULL 
+    THEN 'Not Stated'
+    ELSE CONCAT(CAST(_start_date AS DATE),' ','-',' ','No End Date') 
+  END AS _campaign_date_range,
+  airtable_info.* EXCEPT(_campaignid,_campaignname)
+FROM (
```
Comment (izztsfn) [inline Clients/Carenet/Normalization Script/Ads/carenet_health_all_ads_perf.sql:326]: Create a new CTE to remove subquery here

File: Clients/Carenet/Normalization Script/Ads/carenet_health_all_ads_perf.sql [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+  -- JOIN `carenet_health_mysql.db_airtable_6sense_campaign` c ON CAST(c._campaignid AS STRING)  = CAST(a._6senseid AS STRING)
+  WHERE _datatype = 'Campaign'
+  QUALIFY ROW_NUMBER() OVER(
+    PARTITION BY _6sense._6senseid, _6sense._date 
+    ORDER BY CAST(PARSE_DATE('%m/%e/%Y', _6sense._date) AS TIMESTAMP) DESC
+  ) = 1
+),
+_6sense AS (
+  SELECT *
+  FROM (
```
Comment (izztsfn) [inline Clients/Carenet/Normalization Script/Ads/carenet_health_all_ads_perf.sql:176]: Remove this subquery

File: Clients/Carenet/Normalization Script/Ads/carenet_health_all_ads_perf.sql [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+    ON campaigns.campaign_group_id = campaign_group.groupID
+  WHERE _timestamp IS NOT NULL
+  QUALIFY ROW_NUMBER() OVER (
+    PARTITION BY LI_campaign._timestamp, campaigns._campaignID 
+    ORDER BY LI_campaign._timestamp DESC
+  ) = 1
+),
+_linkedin AS (
+  SELECT * 
+  FROM (
```
Comment (izztsfn) [inline Clients/Carenet/Normalization Script/Ads/carenet_health_all_ads_perf.sql:303]: Remove subquery

File: Clients/Carenet/Normalization Script/Ads/carenet_health_all_ads_perf.sql [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+    CAST(NULL AS INT64) AS _ad_group_id,
+    CAST(_6sense._6senseid AS INT64) AS _ad_id, 
+    -- _adgroup AS _ad_group_name,
+    _6sense._name AS _ad_group_name, 
+    _campaignname AS _campaign_name, 
+    -- _advariation AS _ad_name,
+    '' AS _ad_name,
+    CASE 
+      WHEN _6sense._startdate = '-' 
+      THEN NULL
```
Comment (izztsfn) [inline Clients/Carenet/Normalization Script/Ads/carenet_health_all_ads_perf.sql:94]: Ensure the CASE statement syntax is consistent for the script. Refer the following:
```
CASE xxx
  WHEN xxx THEN xxx
  ELSE xx
END AS _xxx
```

Update the remaining CASE statements to follow the same syntax

**Review by izztsfn — CHANGES_REQUESTED** [review]
Some formatting changes to make. Otherwise pretty decently formatted already :D

**Comment by izztsfn** [comment]
Also @raaayden be sure to update your progress for these scripts in the scheduled script tracker: https://p423itcaltn.sg.larksuite.com/base/MJyebpBYDaX3IKsuY0Ulhwv3gZb?table=tblKzo8GrW53mUQF&view=vewDViB0he

**Comment by raaayden** [comment]
hello @izztsfn , thanks for the review, i have already updated the script !

**Review by izztsfn — APPROVED** [review]
Looks good! Approved

**Pull request merged**